middleware/reverse/network.go

```diff
@@ -0,0 +1,122 @@
+package reverse
+
+import (
+	"net"
+	"regexp"
+	"bytes"
+	"strings"
+)
+
+type network struct {
+	IPnet        *net.IPNet
+	Domain       string
+	Template     string
+	TTL          uint32
+	RegexMatchIP *regexp.Regexp
+}
+
+const hexDigit = "0123456789abcdef"
+const templateNameIP = "{ip}"
+const regexMatchV4 = "((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\-){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))"
```

Comment:
impl question: the middleware knows how that generated names look; why do we need regexp matching then?